### PR TITLE
Expose HealthyPanicThreshold in Helm Chart 

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Description

Support configuration of the healthy_panic_threshold

# Context

This is a follow up to https://github.com/solo-io/gloo/pull/3891. That PR set the healthy panic threshold to 0 (disabling panic mode) in e2e tests. This PR adds support for Kube2e tests.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/3747

### CODE REVIEW

File: test/services/envoy.go [hunk 1]
```diff
@@ -69,14 +68,19 @@ func (ei *EnvoyInstance) buildBootstrapFromConfig(configFile string) (string, er
 	if addr != "" && addr != "127.0.0.1" {
 		config = strings.ReplaceAll(config, "127.0.0.1", addr)
 	}
-
```
Comment (sam-heilbron) [inline test/services/envoy.go:72]: This was added by https://github.com/solo-io/gloo/pull/3891. It was needed so we could make requests to the envoy admin endpoint to set the healthy panic threshold to 0. Now we just set it in the bootstrap configuration, so a request is no longer necessary. Remove the old code.

File: install/helm/gloo/generate/values.go [hunk 2]
```diff
@@ -272,6 +272,7 @@ type GatewayProxy struct {
 	PodTemplate                    *GatewayProxyPodTemplate     `json:"podTemplate,omitempty"`
 	ConfigMap                      *GatewayProxyConfigMap       `json:"configMap,omitempty"`
 	GlobalDownstreamMaxConnections uint32                       `json:"globalDownstreamMaxConnections,omitempty" desc:"the number of concurrent connections needed. limit used to protect against exhausting file descriptors on host machine"`
+	HealthyPanicThreshold          int8                         `json:"healthyPanicThreshold,omitempty" desc:"the percentage of healthy hosts required to load balance based on health status of hosts"`
```
Comment (sam-heilbron) [inline install/helm/gloo/generate/values.go:275]: https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/cluster.proto#envoy-v3-api-field-config-cluster-v3-cluster-commonlbconfig-healthy-panic-threshold HealthyPanicThreshold technically is a Percent (which accepts a double under the hood). But it makes a note that the value will be truncated to the nearest 1%, so I opted to just make this an int.

File: test/kube2e/glooctl/glooctl_suite_test.go [hunk 3]
```diff
@@ -55,14 +56,37 @@ func StartTestHelper() {
...
 	// Install Gloo
-	err = testHelper.InstallGloo(ctx, helper.GATEWAY, 5*time.Minute)
+	err = testHelper.InstallGloo(ctx, helper.GATEWAY, 5*time.Minute, helper.ExtraArgs("--values", valuesOverrideFile))
 	Expect(err).NotTo(HaveOccurred())
 
 	// Check that everything is OK
 	kube2e.GlooctlCheckEventuallyHealthy(1, testHelper, "90s")
 }
 
+func getHelmValuesOverrideFile() (filename string, cleanup func()) {
```
Comment (sam-heilbron) [inline test/kube2e/glooctl/glooctl_suite_test.go:71]: Even though each test suite uses similar functionality (some configured helm override) I opted to define this function on each test suite independently. All this does is write to a file and defer clean up, so I thought the code duplication warranted keeping the test suites isolates and easier to follow. Happy to put a utility function to write to a file and defer cleanup that each of the suites can use if that's preferred though.

File: changelog/v1.6.0-beta14/healthy-panic-threshold-configuration.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+changelog:
+  - type: HELM
+    issueLink: https://github.com/solo-io/gloo/issues/3747
+    description: >
+      Panic mode allows Envoy load balancing to disregard host's health status.
+      With panic mode disabled, we can be more certain that our tests are validating healthy hosts properly.
```
Comment (kdorosh) [inline changelog/v1.6.0-beta14/healthy-panic-threshold-configuration.yaml:6]: I wouldn't mention the tests part in the helm changelog. perhaps in a non user facing one though

Comment (sam-heilbron) [reply changelog/v1.6.0-beta14/healthy-panic-threshold-configuration.yaml:10]: Goood call. Updated the changelog to reflect what I think you were asking for. Let me know if I misinterpreted.

Comment (kdorosh) [reply changelog/v1.6.0-beta14/healthy-panic-threshold-configuration.yaml:10]: looks good to me

**Review by kdorosh — APPROVED** [review]

**Pull request merged**